### PR TITLE
add/modify subject new/edit views

### PR DESCRIPTION
**Pull request by kagoromo**

### CODE REVIEW

**Comment by nguyenngocha** [comment]
OK

File: app/controllers/supervisor/subjects_controller.rb [hunk 1]
```diff
@@ -4,16 +4,17 @@ class Supervisor::SubjectsController < ApplicationController
...
-    subject.tasks.build
-    @subject_form = SubjectForm.new subject
   end
 
   def show
   end
 
+  def new
+    @subject.tasks.build
+    @subject_form = SubjectForm.new @subject
```
Comment (yongsokheng) [inline app/controllers/supervisor/subjects_controller.rb]: Nen tach thanh mot ham de dung cho cho khac

File: app/views/trainee_subjects/show.html.erb [hunk 2]
```diff
@@ -44,6 +44,32 @@
       <% end %>
     </div>
   </div>
+  
+  <div class="col-md-6 col-md-offset-3">
+    <div class="panel panel-default">
+      <div class="panel-heading text-center">
+        <% if @bottom == Settings.activity.my_activities %>
```
Comment (nguyenngocha) [inline app/views/trainee_subjects/show.html.erb]: refactor

File: app/views/trainee_subjects/show.html.erb [hunk 3]
```diff
@@ -44,6 +44,32 @@
...
+          <h3><%= t "activity.all_activities" %></h3>
+        <% else %>
+          <h3><%= t "activity.my_activities" %></h3>
+        <% end %>
+      </div>
+      <div class="panel-body">
+        <ul class="list-unstyled">
+          <% if @activities.any? %>
```
Comment (nguyenngocha) [inline app/views/trainee_subjects/show.html.erb]: refactor

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -21,4 +21,8 @@ def link_to_add_fields name, f, association
     link_to name, "#", class: "add_fields",
       data: {id: id, fields: fields.gsub("\n", "")}
   end
+  
+  def load_user_course course
+    course.user_courses.find_by user: current_user
```
Comment (yongsokheng) [inline app/helpers/application_helper.rb]: Nen viet trong Scope

File: app/views/trainee_subjects/show.html.erb [hunk 5]
```diff
@@ -44,6 +44,32 @@
...
+            <%= render @activities %>
```
Comment (yongsokheng) [inline app/views/trainee_subjects/show.html.erb]: <%= render @activities if @activities.any? %>

File: app/controllers/supervisor/subjects_controller.rb [hunk 6]
```diff
@@ -1,35 +1,36 @@
...
-    subject = Subject.new
-    subject.tasks.build
-    @subject_form = SubjectForm.new subject
   end
 
   def show
   end
 
+  def new
+    byebug
```
Comment (yongsokheng) [inline app/controllers/supervisor/subjects_controller.rb]: bo byebug

File: app/models/course.rb [hunk 7]
```diff
@@ -36,6 +36,10 @@ def build_course_subjects
     end
   end
 
+  def load_user_course user
+    user_courses.find_by user: user
```
Comment (yongsokheng) [inline app/models/course.rb]: nen dung scope
`scope :load_user_course, ->user{user_courses.find_by: user}`

File: app/models/course.rb [hunk 8]
```diff
@@ -36,6 +36,10 @@ def build_course_subjects
     end
   end
 
+  def load_user_course_of user
+    user_courses.find_by user: user
```
Comment (yongsokheng) [inline app/models/course.rb]: Nen viet thanh scope trong user_courses:
`scope :load_course, ->course{find_by course_id: course}`
Khi goi:
`current_user.user_courses.load_course ...`

File: app/controllers/supervisor/subjects_controller.rb [hunk 9]
```diff
@@ -1,31 +1,29 @@
 class Supervisor::SubjectsController < ApplicationController
   load_and_authorize_resource
-  before_action :load_subject_form, only: [:edit, :update]
+  before_action :load_subject_form, only: [:new, :create, :edit, :update]
```
Comment (tranduytung) [inline app/controllers/supervisor/subjects_controller.rb]: dùng expect

File: app/views/supervisor/subjects/_form.html.erb [hunk 10]
```diff
@@ -1,17 +1,26 @@
...
+  </div>
+  <div class="form-group">
+    <%= f.submit t("subject.submit"),
+      class: "btn btn-primary col-md-4 col-md-offset-4" %>
+  </div>
+</div>
+<br>
+<div class="form-group">
+  <%= link_to t("subject.go_back"),
+    supervisor_subjects_path, class: "btn btn-default col-md-4 col-md-offset-4" %>
```
Comment (tranduytung) [inline app/views/supervisor/subjects/_form.html.erb]: quá 80 kí tự

File: app/views/supervisor/subjects/_subject.html.erb [hunk 11]
```diff
@@ -1,7 +1,24 @@
...
+<div class="row">
+  <div class="col-md-8 col-md-offset-1">
+    <h3>
+      <%= link_to subject.content, supervisor_subject_path(subject) %>
+    </h3>
+    <p><%= subject.description %></p>
+  </div>
+  <div class="col-md-2">
+    <div class="form-group">
+      <br><%= link_to t("subject.view_subject"), supervisor_subject_path(subject),
```
Comment (tranduytung) [inline app/views/supervisor/subjects/_subject.html.erb]: Quá 80 kí tự

File: app/views/trainee_subjects/show.html.erb [hunk 12]
```diff
@@ -44,6 +44,27 @@
...
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
 </div>
-<%= render @activities %>
-<%= paginate @activities %>
+
+
```
Comment (tranduytung) [inline app/views/trainee_subjects/show.html.erb]: 2 enter

**Comment by yongsokheng** [comment]
OK

**Pull request merged**